linkerd/opaq-route/src/match.rs

```diff
@@ -0,0 +1,21 @@
+use std::cmp::Ordering;
+
+/// Matches TCP sessions. For now, this is a placeholder
+#[derive(Clone, Debug, Default, Hash, PartialEq, Eq)]
+pub struct MatchSession(());
```

Comment:
I suspect this is coming from the TLSRoute type--do we expect TCP routes to use "session" terminology?

Reply:
Should we think of a better name. `MatchTCPConnection` ? Or try to get rid of this whole machinery here. And yes, this is coming from the TLSRoute type.

Reply:
This makes sense indeed. Will merge and update control plane to return only one route always.